runtime/src/main/java/org/corfudb/runtime/collections/StreamingManager.java

```diff
@@ -0,0 +1,147 @@
+package org.corfudb.runtime.collections;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.StreamSubscriptionException;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A streaming subscription manager that allows clients to listen on
+ * the transaction updates of interested tables. The updates will be
+ * streamlined and clients can get notifications via the registered
+ * call backs.
+ * <p>
+ * Created by WenbinZhu on 11/5/20.
+ */
+@Slf4j
+public class StreamingManager {
+
+    // Number of thread in polling and notification pool.
+    private static final int NUM_THREAD_PER_POOL = 4;
+
+    // Default buffer size for each subscription.
+    private static final int DEFAULT_BUFFER_SIZE = 50;
```

Comment:
These have to be configurable and exposed. We have already seen bugs where we need to be able to tune a bunch of parameters. Unfortunately, most of our configurations are not configurable and require source code changes.

Reply:
The buffer size is already exposed in the subscribe() API, clients can specify a customed buffer size, or not to use this default setting.